### PR TITLE
Allow enabling individual experimental features

### PR DESCRIPTION
**Pull request by ilevkivskyi**
Ref #13685

### CODE REVIEW

**Comment by ilevkivskyi** [comment]
Hm it looks like this causes a crash, I will try to fix.

**Comment by ilevkivskyi** [comment]
Hm, the second crash shows that we can't probably say `Unpack` support is complete yet. We should probably at least support `*args: Unpack[tuple[int, str, int]]` (supporting unions will be tricky, and we don't support unions of TypedDicts anyway, and unions are not specified by the PEP).

**Comment by ilevkivskyi** [comment]
Btw a simple repro for the crash is
```python
class E(Generic[T]):
    def __init__(self, *args: Unpack[tuple[int, str]]) -> None: ...
E()
```

Also while playing with this I have found two bugs in the `**kwds` support for `Unpack`. First, this crashes:
```python
class C:
    def __init__(self, **kwds: Unpack) -> None: ...
```
Instead we should give an error about missing type arg. Second, this doesn't substitute implicit `Any` for a generic TypedDict, causing weird error:
```python
T = TypeVar("T")
class TD(TypedDict, Generic[T]):
    x: T
    y: int

def foo(**kwds: Unpack[TD]) -> None: ...  # Should be same as `TD[Any]`
foo(x="yes", y=42)
```

File: mypy/options.py [hunk 1]
```diff
@@ -60,6 +60,10 @@ class BuildType:
     "debug_cache"
 }
 
+# Features that are currently incomplete/experimental
+TYPE_VAR_TUPLE = "TypeVarTuple"
+INCOMPLETE_FEATURES = {TYPE_VAR_TUPLE}
```
Comment (sobolevn) [inline mypy/options.py:65]:
```suggestion
TYPE_VAR_TUPLE: Final = "TypeVarTuple"
INCOMPLETE_FEATURES: Final = frozenset((TYPE_VAR_TUPLE,))
```

**Review by sobolevn — COMMENTED** [review]
Looks cool!

Question: can we somehow bind feature ids to error codes?
So users won't have to memorize two separate things and their mappings?

**Comment by ilevkivskyi** [comment]
> can we somehow bind feature ids to error codes?

No, not really. Many features (in particular those that are relevant here like `Unpack` and `TypeVarTuple`) don't have any errors related to them. This PR is specifically for such cases.

**Pull request merged**